custom_components/tuya_local/devices/ble_smart_plant_moisture.yaml

```diff
@@ -0,0 +1,48 @@
+name: Moisture sensor
+products:
+  - id: bfbf6c7bb11573ed23j4ya
+    name: SGS01
+primary_entity:
+  entity: sensor
+  name: Soil Moisture
+  class: humidity
+  dps:
+    - id: 3
+      name: sensor
+      type: integer
+      force: true
+      unit: "%"
+secondary_entities:
+  - entity: sensor
+    name: Soil Temperature
+    class: temperature
+    dps:
+      - id: 5
+        name: sensor
+        type: integer
+        mapping:
+          - scale: 10
+        unit: C
```

Comment:
What happens if temperature unit is changed? Does the device start reporting in F? If so, then the unit dp needs to be added to this entity rather than specifying a fixed unit here.

Reply:
Okay, I will delete it. Despite what is stated in the document, it doesn't work.

```
{
                    "abilityId": 9,
                    "accessMode": "rw",
                    "code": "temp_unit_convert",
                    "description": "",
                    "extensions": {
                        "iconName": "icon-dp_mode",
                        "attribute": "1152"
                    },
                    "name": "温标切换",
                    "typeSpec": {
                        "range": ["c", "f"],
                        "type": "enum",
                        "typeDefaultValue": "c"
                    }
                }
```

Reply:
@make-all Without this parameter (DP 9), the device cannot be added, but as mentioned earlier, changing the unit of measurement does not lead to anything. What if the parameter is needed but needs to be hidden?

Reply:
How did you determine that the change did not work?

The UI converts temperatures according to the user preferences, so if the device reports 20C and units are set to C, it will show 20C, but if the device reports 68F, it will also show 20C.

But a missing DP should not prevent a device from being added. It might make a different config look like a better match  than the actual config you want, but this config should still be listed, even if it is not selected by default.

Reply:
I based it on the Tuya documentation. There is a DP for changing the temperature unit of measurement. However, it is not possible to change the unit of measurement within the Tuya Smart application as there is no such option available. I make the change in tuya-local, but after a minute, the parameter returns to "Celsius".

There was nothing better than my converter, so it simply refused to be added without that DP.